Flaky E2E: error: timed out waiting for the condition on deployments/prometheus-operator

<!--
Please use this template for reporting flaky tests.
Links to specific failures in Prow are appreciated.
-->

**Which test is flaking?**:
pull-kueue-test-e2e-main-1-35

**Link to failed CI job or steps to reproduce locally**:
https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_kueue/9985/pull-kueue-test-e2e-main-1-35/2051647188466929664

**Failure message or logs**:
```
 clusterrolebinding.rbac.authorization.k8s.io/prometheus-operator serverside-applied
clusterrole.rbac.authorization.k8s.io/prometheus-operator serverside-applied
deployment.apps/prometheus-operator serverside-applied
serviceaccount/prometheus-operator serverside-applied
service/prometheus-operator serverside-applied
error: timed out waiting for the condition on deployments/prometheus-operator
Switched to context "kind-kind".
Exporting logs for cluster "kind" to:
/logs/artifacts/run-test-e2e-singlecluster-1.35.0
No resources found in kueue-system namespace.
Deleting cluster "kind" ...
Deleted nodes: ["kind-worker2" "kind-worker" "kind-control-plane"]
make: *** [Makefile-test.mk:212: run-test-e2e-singlecluster-1.35.0] Error 1
+ EXIT_VALUE=2
+ set +o xtrace
Cleaning up after docker in docker.
================================================================================
Waiting 30 seconds for pods stopped with terminationGracePeriod:30

```

**Anything else we need to know?**:

--- a/hack/testing/e2e-common.sh
+++ b/hack/testing/e2e-common.sh
@@ -224,6 +224,8 @@ fi
 
 if [[ -n "${PROMETHEUS_OPERATOR_VERSION:-}" ]]; then
     export PROMETHEUS_OPERATOR_BUNDLE="https://github.com/prometheus-operator/prometheus-operator/releases/download/${PROMETHEUS_OPERATOR_VERSION}/bundle.yaml"
+    export PROMETHEUS_OPERATOR_IMAGE="quay.io/prometheus-operator/prometheus-operator:${PROMETHEUS_OPERATOR_VERSION}"
+    export PROMETHEUS_CONFIG_RELOADER_IMAGE="quay.io/prometheus-operator/prometheus-config-reloader:${PROMETHEUS_OPERATOR_VERSION}"
 fi
 
 if [[ -n "${DRA_EXAMPLE_DRIVER_VERSION:-}" ]]; then
@@ -515,6 +517,10 @@ function prepare_docker_images {
     if [[ -n ${SPARKOPERATOR_VERSION:-} && ("$GINKGO_ARGS" =~ feature:spark || ! "$GINKGO_ARGS" =~ "--label-filter") ]]; then
         e2e_docker_pull_if_needed "${SPARKOPERATOR_IMAGE}"
     fi
+    if [[ -n ${PROMETHEUS_OPERATOR_VERSION:-} && ("$GINKGO_ARGS" =~ feature:prometheus || ! "$GINKGO_ARGS" =~ "--label-filter") ]]; then
+        e2e_docker_pull_if_needed "${PROMETHEUS_OPERATOR_IMAGE}"
+        e2e_docker_pull_if_needed "${PROMETHEUS_CONFIG_RELOADER_IMAGE}"
+    fi
 }
 
 # $1 cluster
@@ -579,7 +585,7 @@ function kind_load {
         install_cert_manager "${e2e_kubeconfig}"
     fi
     if [[ -n ${PROMETHEUS_OPERATOR_VERSION:-} && ("$GINKGO_ARGS" =~ feature:prometheus || ! "$GINKGO_ARGS" =~ "--label-filter") ]]; then
-        install_prometheus_operator "${e2e_kubeconfig}"
+        install_prometheus_operator "${e2e_cluster_name}" "${e2e_kubeconfig}"
     fi
     if [[ -n ${CLUSTERPROFILE_VERSION:-} ]]; then
         install_multicluster "${e2e_kubeconfig}"
@@ -1135,9 +1141,11 @@ function install_cert_manager {
     kubectl wait --kubeconfig="${kubeconfig}" deploy/cert-manager-cainjector -n "${ns}" --for=condition=available --timeout=5m || true
 }
 
-# $1 kubeconfig option
+# $1 cluster name
+# $2 kubeconfig option
 function install_prometheus_operator {
-    local kubeconfig=${1:-}
+    local name=$1
+    local kubeconfig=${2:-}
     local ns="default"
     local deployment_name="prometheus-operator"
     local expected_version="${PROMETHEUS_OPERATOR_VERSION:-}"
@@ -1163,6 +1171,8 @@ function install_prometheus_operator {
         fi
     fi
 
+    cluster_kind_load_image "${name}" "${PROMETHEUS_OPERATOR_IMAGE}"
+    cluster_kind_load_image "${name}" "${PROMETHEUS_CONFIG_RELOADER_IMAGE}"
     kubectl apply --kubeconfig="${kubeconfig}" --server-side -f "${PROMETHEUS_OPERATOR_BUNDLE}"
     kubectl wait deploy/"${deployment_name}" -n "${ns}" \
         --for=condition=available --timeout=5m --kubeconfig="${kubeconfig}"

